src/Passes/Source/Passes/TransformationRule/LibTransformationRule.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Passes/TransformationRule/TransformationRule.hpp"
+#include "Rules/Factory.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace
+{
+llvm::PassPluginLibraryInfo getTransformationRulePluginInfo()
```

Comment:
I seem to be missing something here; neither the folder structure nor the naming within this file gives me any indication that this is specific to a particular profile, even though the source code seems to implement a particular set of rules that would apply for specific profile configurations only. Is this code expected to change in the future and if so, how?

Reply:
This is just an example pass that can be used with `opt`. In general, all files that starts with `Lib{name}` under passes implement a loadable pass. As we cannot (at least I believe so) provide a pass configuration except for O1, O2 or O3, the logic is hard coded. We may choose to to make multiple passes using the rule engine in the future, or we may choose to deprecate this alltogether.